### PR TITLE
[FTE][fling-tutorial][TV][2.5] Button focus fails in online tutorial because of no permission to document.registerElement method

### PR DESCRIPTION
**Pull request by Fischer-L**
This patch:
- Remove the components/smart-button because in the production build, the app is 'app://' protocol and  the iframe is 'http://' protocol, even the app claims the web-component permission but the iframe still cannot get the web-component permission.
  
  As a result, the gaia-components/smart-button which relies on the web-component feature wouldn't function.
  
  p.s: If it was in the engineering build, the iframe would be allowed to get the web-component permission in this case.
- Reimplement the UI styles and behaviours of the gaia-components/smart-button without using document.registerElement

### CODE REVIEW

**Comment by rexboy7** [comment]
Did you discussed this problem with Gecko team? I think this is a kind of workaround. Gecko should be able to support web components properly.

**Comment by Fischer-L** [comment]
This is because the app's protocol (app://) is different from the iframe's protocol (http://).
Update the issue description in the top for a clearer explanation, thanks.

File: js/index.js [hunk 1]
```diff
@@ -5,6 +5,10 @@
 
   var tutorialContainer = document.body;
 
+  var smartButtons = Array.prototype.slice.call(
```
Comment (rexboy7) [inline js/index.js:12]: Please add `XXX: These buttons come from smart-button web components but they are supported only for certified apps in v2.5. We should change them back to web components once we can use them.`

Comment (Fischer-L) [inline js/index.js:12]: Updated, thanks

**Pull request merged**